packages/chatbot-server-mongodb-public/src/config.eval.ts

```diff
@@ -0,0 +1,207 @@
+import { systemPrompt } from "./systemPrompt";
+import {
+  Conversation,
+  generateResponse,
+  logger,
+  Message,
+  ObjectId,
+} from "mongodb-chatbot-server";
+import {
+  getContextsFromUserMessage,
+  getLastAssistantMessageFromMessages,
+  getLastUserMessageFromMessages,
+  JUDGE_EMBEDDING_MODEL,
+  JUDGE_LLM,
+  OPENAI_API_KEY,
+  OPENAI_API_VERSION,
+  OPENAI_ENDPOINT,
+} from "./test/evalHelpers";
+import { AnswerRelevancy, ContextRelevancy, Faithfulness } from "autoevals";
+import fs from "fs";
+import path from "path";
+import { getConversationsEvalCasesFromYaml } from "./test/getConversationEvalCasesFromYaml";
+import { strict as assert } from "assert";
+
+interface ConversationEvalCaseInput {
+  previousConversation: Conversation;
+  latestMessageText: string;
+}
+interface ConversationEvalCase
+  extends EvalCase<ConversationEvalCaseInput, unknown, unknown> {
+  name: string;
+  input: ConversationEvalCaseInput;
+  tags?: MongoDbTag[];
+}
+
+interface ConversationTaskOutput {
+  assistantMessageContent: string;
+  context?: string[];
+  allowedQuery: boolean;
+}
+
+type ConversationEvalScorer = EvalScorer<
+  ConversationEvalCaseInput,
+  ConversationTaskOutput,
+  unknown
+>;
+
+// -- Evaluation metrics --
+const RetrievedContext: ConversationEvalScorer = async (args) => {
+  args.output.context;
+  return {
+    name: "RetrievedContext",
+    score: args.output.context?.length ? 1 : 0,
+  };
+};
+
+const AllowedQuery: ConversationEvalScorer = async (args) => {
+  return {
+    name: "AllowedQuery",
+    score: args.output.allowedQuery ? 1 : 0,
+  };
+};
+
+function getConversationRagasConfig(
+  scorerArgs: Parameters<ConversationEvalScorer>[0],
+  judgeModelConfig: {
+    model: string;
+    azureOpenAi: {
+      apiKey: string;
+      apiVersion: string;
+      endpoint: string;
+    };
+  }
+) {
+  return {
+    output: scorerArgs.output.assistantMessageContent,
+    context: scorerArgs.output.context,
+    input: scorerArgs.input.latestMessageText,
+    model: judgeModelConfig.model,
+    azureOpenAi: judgeModelConfig.azureOpenAi,
+  };
+}
+
+const judgeModelConfig = {
+  model: JUDGE_LLM,
+  embeddingModel: JUDGE_EMBEDDING_MODEL,
+  azureOpenAi: {
+    apiKey: OPENAI_API_KEY,
+    apiVersion: OPENAI_API_VERSION,
+    endpoint: OPENAI_ENDPOINT,
+  },
+};
+
+const ConversationFaithfulness: ConversationEvalScorer = async (args) => {
+  return Faithfulness(getConversationRagasConfig(args, judgeModelConfig));
+};
+
+// Commenting out until this PR is merged and released https://github.com/braintrustdata/autoevals/pull/94
+// const ConversationAnswerRelevancy: ConversationEvalScorer = async (args) => {
+//   return AnswerRelevancy({
+//     ...getConversationRagasConfig(args, judgeModelConfig),
+//     embeddingModel: JUDGE_EMBEDDING_MODEL,
+//   });
+// };
+
+const ConversationContextRelevancy: ConversationEvalScorer = async (args) => {
+  return ContextRelevancy(getConversationRagasConfig(args, judgeModelConfig));
+};
```

Comment:
I see some errors with this in the Braintrust UI. Any idea what's going on?

e.g. https://www.braintrust.dev/app/mdb-test/p/mongodb-chatbot-conversations/experiments/mongodb-chatbot-latest?r=3fe62b58-3ea1-437b-906c-41a59a7f0aba&s=9fb53e4e-05fd-4d09-99dd-6a978445a451

<img width="753" alt="Screenshot 2024-09-04 at 7 42 37 PM" src="https://github.com/user-attachments/assets/8a2083ab-c3ee-41d5-bf05-96b38a57c6da">

Reply:
LLM rate limits hit (HTTP status `429`). the error is somewhere in the UI, but you have to look around/scroll a bit for it

Reply:
by not running tests in parallel we should be able to avoid this...fairly annoying for spikey workloads like this but ai magic aint infinite (yet)